backend/src/apiserver/resource/resource_manager_util.go

```diff
@@ -207,14 +207,25 @@ func OverrideParameterWithSystemDefault(workflow util.Workflow, apiRun *api.Run)
 	if servercommon.GetBoolConfigWithDefault(HasDefaultBucketEnvVar, false) {
 		patchedSlice := make([]wfv1.Parameter, 0)
 		for _, currentParam := range workflow.Spec.Arguments.Parameters {
-			desiredValue, err := PatchPipelineDefaultParameter(*currentParam.Value)
-			if err != nil {
-				return fmt.Errorf("failed to patch default value to pipeline. Error: %v", err)
+			if currentParam.Value != nil {
+				desiredValue, err := PatchPipelineDefaultParameter(*currentParam.Value)
+				if err != nil {
+					return fmt.Errorf("failed to patch default value to pipeline. Error: %v", err)
+				}
+				patchedSlice = append(patchedSlice, wfv1.Parameter{
```

Comment:
Perhaps we can move this piece of logic out from `if currentParam.Value != nil {...} else {...}` because it's a common logic?

Reply:
A little problem here is that if I move out "patchedSlice = append...(desiredValue)" part of the if-else block, I'll still have to check that this append operation only happens when at least one of the Value and Default is non-nil (i.e., desiredValue is valid). In other words, when both Value and Default are nil, this append operation should be skipped. Therefore,  I figure that moving it out seems adding more code somehow :) So I choose this one, which is a little bit redundant but arguably easy to read though.